### PR TITLE
additions to sphere module

### PR DESCRIPTION
**Pull request by selipot**
Additions to the sphere module:

coriolis_frequency
tangentplane_to_sphere
sphere_to_tangentplane

### CODE REVIEW

**Comment by selipot** [comment]
Argh, I don't know why this PR contains commits related to the documentation :(

**Comment by philippemiron** [comment]
You should be able to sync your main branch, then rebase that beanch with main.

**Comment by milancurcic** [comment]
Don't worry too much about docs commits, we can ignore them during the review.

File: clouddrift/sphere.py [hunk 1]
```diff
@@ -566,3 +568,168 @@ def cartesian_to_spherical(
...
+
+    Parameters
+    ----------
+        latitude : float or np.ndarray
+            Latitude in degrees.
+
+    Returns
+    ------
+        f : float or np.ndarray
+            Signed Coriolis frequency in radian per seconds.
```
Comment (milancurcic) [inline clouddrift/sphere.py:718]:
```suggestion
    latitude : float or np.ndarray
        Latitude in degrees.

    Returns
    ------
    f : float or np.ndarray
        Signed Coriolis frequency in radian per seconds.
```

**Review by milancurcic — APPROVED** [review]

**Review by selipot — COMMENTED** [review]
ok

File: clouddrift/sphere.py [hunk 2]
```diff
@@ -5,10 +5,12 @@
 """
 
 import numpy as np
-from typing import Optional, Tuple
+from typing import Optional, Tuple, Union
 import xarray as xr
+import warnings
 
 EARTH_RADIUS_METERS = 6.3781e6
+EARTH_ROTATION_RATE = 7.2921159e-5
```
Comment (philippemiron) [inline clouddrift/sphere.py:13]: We can get the exact value like this. With the Earth Rotation being 23 hours 56 minutes 4.091 seconds.
```suggestion
EARTH_ROTATION_RATE = 2*np.pi/86164.091
```

File: clouddrift/sphere.py [hunk 3]
```diff
@@ -566,3 +568,162 @@ def cartesian_to_spherical(
...
+        v : float or np.ndarray
+            Second component of Cartesian vector.
+        w : float or np.ndarray
+            Third component of Cartesian vector.
+
+    Raises
+    ------
+    Warning
+        Raised if the input latitude is not in the expected range [-90, 90].
+
```
Comment (philippemiron) [inline clouddrift/sphere.py:679]:
```suggestion
```

File: clouddrift/sphere.py [hunk 4]
```diff
@@ -566,3 +568,162 @@ def cartesian_to_spherical(
...
+
+def tangentplane_to_cartesian(
+    up: Union[float, np.ndarray],
+    vp: Union[float, np.ndarray],
+    longitude: Union[float, np.ndarray],
+    latitude: Union[float, np.ndarray],
+) -> Union[Tuple[float], Tuple[np.ndarray]]:
+    """
+    Return the three-dimensional Cartesian components of a vector conatined in
```
Comment (philippemiron) [inline clouddrift/sphere.py:646]:
```suggestion
    Return the three-dimensional Cartesian components of a vector contained in
```

File: clouddrift/sphere.py [hunk 5]
```diff
@@ -566,3 +568,162 @@ def cartesian_to_spherical(
...
+    a  plane tangent to a spherical Earth.
```
Comment (philippemiron) [inline clouddrift/sphere.py:647]:
```suggestion
    a plane tangent to a spherical Earth.
```

File: clouddrift/sphere.py [hunk 6]
```diff
@@ -566,3 +568,162 @@ def cartesian_to_spherical(
...
+        w : float or np.ndarray
+            Third component of Cartesian vector.
+        longitude : float or np.ndarray
+            Longitude in degrees of tangent point of plane.
+        latitude : float or np.ndarray
+            Latitude in degrees of tangent point of plane.
+
+    Returns
+    -------
+
```
Comment (philippemiron) [inline clouddrift/sphere.py:605]:
```suggestion
```

**Review by philippemiron — APPROVED** [review]

File: clouddrift/sphere.py (hunk 2 above)
Comment (milancurcic) [reply clouddrift/sphere.py:13]:
```suggestion
EARTH_ROTATION_RATE = 7.2921159e-5
EARTH_DAY_SECONDS = 86164.091
EARTH_ROTATION_RATE = 2 * np.pi / EARTH_DAY_SECONDS
```

**Pull request merged**